test/unit/remote_tests.rb

```diff
@@ -35,6 +37,13 @@ class UnitTests < Assert::Context
     end
     subject{ @cmd }
 
+    private
+
+    def ssh_cmd_str(cmd_str, host, args)
+      val = "\"#{cmd_str}\"".gsub("\\", "\\\\\\").gsub('"', '\"')
+      "ssh #{args} #{host} -- \"sh -c #{val}\""
+    end
```

Comment:
@jcredding also I feel kinda sheepish about testing with a blatant rip-off of the live code, but I couldn't think of a better way.  Feel free to suggest something better.